pkg/producer/kafka.go

```diff
@@ -26,39 +31,67 @@ import (
 	"github.com/vmware/go-ipfix/pkg/producer/protobuf"
 )
 
-var (
-	KafkaConfigVersion sarama.KafkaVersion
-)
-
 type KafkaProducer struct {
 	producer             sarama.AsyncProducer
+	enableLogSuccesses   bool
 	topic                string
 	protoSchemaConvertor convertor.IPFIXToKafkaConvertor
 }
 
-func NewKafkaProducer(asyncProducer sarama.AsyncProducer, topic string, schemaType string) *KafkaProducer {
+type ProducerInput struct {
+	// KafkaBrokers is a string of addresses of Kafka broker systems
+	KafkaBrokers         []string
+	KafkaVersion         sarama.KafkaVersion
+	KafkaTopic           string
+	KafkaProtoSchema     string
+	KafkaTLSEnabled      bool
+	KafkaCAFile          string
+	KafkaTLSCertFile     string
+	KafkaTLSKeyFile      string
+	KafkaTLSSkipVerify   bool
+	KafkaLogErrors       bool
+	KafkaLogSuccesses    bool
+	EnableSaramaDebugLog bool
+}
+
+func NewKafkaProducer(asyncProducer sarama.AsyncProducer, enableLogSuccesses bool, topic string, schemaType string) *KafkaProducer {
 	return &KafkaProducer{
 		producer:             asyncProducer,
+		enableLogSuccesses:   enableLogSuccesses,
 		topic:                topic,
 		protoSchemaConvertor: convertor.ProtoSchemaConvertor[schemaType](),
 	}
 }
 
 // InitKafkaProducer with broker addresses and other Kafka config parameters.
-func InitKafkaProducer(addrs []string, topic string, protoSchema string, logErrors bool) (*KafkaProducer, error) {
+func InitKafkaProducer(input ProducerInput) (*KafkaProducer, error) {
 	kafkaConfig := sarama.NewConfig()
-	kafkaConfig.Version = KafkaConfigVersion
-	kafkaConfig.Producer.Return.Successes = false
-	kafkaConfig.Producer.Return.Errors = logErrors
+	kafkaConfig.Version = input.KafkaVersion
+	kafkaConfig.Producer.Return.Successes = input.KafkaLogSuccesses
+	kafkaConfig.Producer.Return.Errors = input.KafkaLogErrors
 
-	asyncProducer, err := sarama.NewAsyncProducer(addrs, kafkaConfig)
+	// Initialize TLS certificates
+	if input.KafkaTLSEnabled {
+		if tlsConfig := setupTLSConfig(input.KafkaCAFile, input.KafkaTLSCertFile, input.KafkaTLSKeyFile); tlsConfig != nil {
+			kafkaConfig.Net.TLS.Config = tlsConfig
+			kafkaConfig.Net.TLS.Enable = true
+			if input.KafkaTLSSkipVerify {
```

Comment:
Seems that `input.KafkaTLSSkipVerify` is not used other places other than here. Do we need to handle the case when `input.KafkaTLSSkipVerify` is set?

Reply:
Got it. We can add code to ignore secure checking with proper reasoning description like:
```
// #nosec G402: only applicable for testing purpose
t.InsecureSkipVerify = skipVerify
```

Reply:
Why didn't we pass input.KafkaTLSSkipVerify to setupTLSConfig(), which returns a tls.Config object?
I thought that value can be directly assigned to the returning tls.Config, such as 
tls.Config{InsecureSkipVerify: input.KafkaTLSSkipVerify} ?

Reply:
golang lint test fails if we directly the value like above. However, I moved the assignment to `setupTLSConfig`.

Sorry for late response. I had put this PR on hold. Please take a look again.